astropy/table/pprint.py

```diff
@@ -384,8 +383,11 @@ def _pformat_col_iter(self, col, max_lines, show_name, show_unit, outs,
         n_print2 = max_lines // 2
         n_rows = len(col)
 
-        col_format = col_getattr(col, 'format')
-        format_func = _format_funcs.get(col_format, _auto_format_func)
+        col_format = col.info.format
+        if col_format is None and hasattr(col.info, 'default_format_func'):
```

Comment:
@mhvk - I had not read your proposed code carefully before, but was thinking about this offline now and then over the last few days (busy with other stuff).  I basically wrote the same code in my head, so we are agreed!